src/D2L.Bmx/OktaAuthenticator.cs

```diff
@@ -99,35 +105,64 @@ private bool TryAuthenticateFromCache(
 		return true;
 	}
 
-	private async Task<IOktaAuthenticatedClient?> GetDssoAuthenticatedClientAsync(
-		Uri orgUrl,
-		string user,
+	private record DssoAuthOptions( string BrowserPath, bool BypassBrowserSecurity, bool PrintDebugMessage );
+
+	private bool TryGetDssoAuthOptions(
 		bool nonInteractive,
-		bool bypassBrowserSecurity
+		bool bypassBrowserSecurity,
+		[NotNullWhen( returnValue: true )] out DssoAuthOptions? options
 	) {
+		options = null;
+
+		bool printDebugMessage =
+			BmxEnvironment.IsDebug
+			|| (
+				// if the user passed --experimental-bypass-browser-security, they've explicitly expressed interest in the browser/DSSO process
+				bypassBrowserSecurity
```

Comment:
Aside:
I'm actually wondering if we can afford to not have this flag at all, and just always use `--no-sandbox` when elevated.
BMX has always trusted the org Okta URL. I mean, it sends the user's Okta password (a super sensitive piece of data) to that URL after all, and it never had any warnings about that.
Seems weird that we're now asking users to confirm that they trust their own Okta org when we're just visiting the org homepage.

Reply:
It's an additional risk to the user's machine, right? Seems like it's worth keeping because of that

Reply:
The question is more that - under what threat model & scenario is it not ok/too risky to load the URL in an unsandboxed browser, but at the same time ok to send the user's Okta password to that very same URL (modulo path difference)?

I think the flag only makes sense if we can find a non-null answer to the question above, and I'm having trouble finding a reasonable threat model & scenario that meets the criteria.

Reply:
Coming around to the idea, especially since it reduces the mental model re: number of states BMX can be in.